doc/developers/contributing.rst

```diff
@@ -790,17 +790,66 @@ In the following example, k is deprecated and renamed to n_clusters::
 
     import warnings
 
-    def example_function(n_clusters=8, k=None):
-        if k is not None:
+    def example_function(n_clusters=8, k='deprecated'):
+        if k != 'deprecated':
```

Comment:
Speaking with @lesteve, it is true that this line look really weird in english:
"if k is not deprecated then warn for deprecation"

Reply:
@glemaitre @lesteve 
I think ``deprecated``, along with ``warn`` can be regarded as symbols here. ``k != 'deprecated'`` can be regarded as : if you use the deprecated parameter, then we 'll raise DeprecationWarning.
Revert back to ``None`` if you don't like it?

Reply:
I think ``warn`` makes sense for changed behaviors, because then it's
```python
if value == "warn":
   # do the warning
```
as below. we could have "not specified" for removing a parameter as a symbol.
```python
if value != "not specified":
   # do the warning
```
That's less clear in the signature, though.

Reply:
maybe "not used" would be better?